openstack/baremetal/v1/nodes/requests.go

```diff
@@ -478,3 +479,89 @@ func ChangePowerState(client *gophercloud.ServiceClient, id string, opts PowerSt
 	})
 	return
 }
+
+// This is the desired RAID configuration on the bare metal node.
+type TargetRAIDConfig struct {
+	LogicalDisks []LogicalDisk `json:"logical_disks"`
+}
+
+// TargetRAIDConfigBuilder allows extensions to modify a set RAID config request.
+type TargetRAIDConfigBuilder interface {
+	ToRAIDConfigMap() (map[string]interface{}, error)
+}
+
+type LogicalDisk struct {
+	// Size (Integer) of the logical disk to be created in GiB.  If unspecified, "MAX" will be used.
+	SizeGB *int `json:"size_gb"`
+
+	// RAID level for the logical disk.
+	RAIDLevel string `json:"raid_level" required:"true"`
```

Comment:
Up to you if you wanted to create a custom type for the RAID level (since there is a finite amount of choices)